go/vt/vtgate/evalengine/testcases/cases.go

```diff
@@ -1436,6 +1437,37 @@ func FnTrim(yield Query) {
 	}
 }
 
+func FnSubstr(yield Query) {
+	mysqlDocSamples := []string{
```

Comment:
we should add a couple of `NULL`s and maybe a couple of tests with numerical types as well